src/v5_hal/test/pathing/test_PathPoint.cpp

```diff
@@ -0,0 +1,118 @@
+            firstPoint = new PathPoint(0, Pose(Eigen::Vector2d(0, 0), Eigen::Rotation2Dd(0)), Eigen::Vector2d(1, 0), 0);
+            secondPoint = new PathPoint(1, Pose(Eigen::Vector2d(1, 0), Eigen::Rotation2Dd(0)), Eigen::Vector2d(1, 0), 0);
+        }
+
+        virtual void TearDown() {
+            delete firstPoint;
+            delete secondPoint;
+        }
+};
+
+class yChangeConstantVelModule : public ::testing::Test {
+    protected:
+        PathPoint* firstPoint;
+        PathPoint* secondPoint;
+
+        virtual void SetUp() {
+            // Creates two points that only have a positive change in y with constant velocity
+            firstPoint = new PathPoint(0, Pose(Eigen::Vector2d(0, 0), Eigen::Rotation2Dd(0)), Eigen::Vector2d(0, 1), 0);
+            secondPoint = new PathPoint(1, Pose(Eigen::Vector2d(0, 1), Eigen::Rotation2Dd(0)), Eigen::Vector2d(0, 1), 0);
+        }
+
+        virtual void TearDown() {
+            delete firstPoint;
+            delete secondPoint;
+        }
+};
+
+class xyChangeConstantVelModule : public ::testing::Test {
+    protected:
+        PathPoint* firstPoint;
+        PathPoint* secondPoint;
+
+        virtual void SetUp() {
+            // Creates two points that have a positive change in x and y with constant velocity
+            firstPoint = new PathPoint(0, Pose(Eigen::Vector2d(0, 0), Eigen::Rotation2Dd(0)), Eigen::Vector2d(1, 1), 0);
+            secondPoint = new PathPoint(1, Pose(Eigen::Vector2d(1, 1), Eigen::Rotation2Dd(0)), Eigen::Vector2d(1, 1), 0);
+        }
+
+        virtual void TearDown() {
+            delete firstPoint;
+            delete secondPoint;
+        }
+};
+
+class noChangeModule : public ::testing::Test {
+    protected:
+        PathPoint* firstPoint;
+        PathPoint* secondPoint;
+
+        virtual void SetUp() {
+            // Creates two points that do not change
+            firstPoint = new PathPoint(0, Pose(Eigen::Vector2d(0, 0), Eigen::Rotation2Dd(0)), Eigen::Vector2d(0, 0), 0);
+            secondPoint = new PathPoint(1, Pose(Eigen::Vector2d(0, 0), Eigen::Rotation2Dd(0)), Eigen::Vector2d(0, 0), 0);
+        }
+
+        virtual void TearDown() {
+            delete firstPoint;
+            delete secondPoint;
+        }
+};
+
+TEST_F(xChangeConstantVelModule, xChangeConstantVel) {
+    PathPoint result = firstPoint->interpolateTo(*secondPoint, 1);
+
+    EXPECT_EQ(secondPoint->getLinearVelocity().x(), result.getLinearVelocity().x());
+    EXPECT_EQ(secondPoint->getLinearVelocity().y(), result.getLinearVelocity().y());
+    EXPECT_EQ(secondPoint->getPose().position.x(), result.getPose().position.x());
+    EXPECT_EQ(secondPoint->getPose().position.y(), result.getPose().position.y());
+    EXPECT_EQ(secondPoint->getPose().angle.angle(), result.getPose().angle.angle());
+    EXPECT_EQ(secondPoint->getRotationalVelocity(), result.getRotationalVelocity());
+    EXPECT_EQ(secondPoint->getTime(), result.getTime());
+}
+
```

Comment:
We should probably add block comments above these tests with a short description of what they are actually testing

Reply:
A common format for this comment and just structure for tests in general is called given, when, then. Given is the inputs required to setup the test, when is an action that is taken to cause inputs to be turned into outputs, and then the then is the outputs of whatever action is taken.

I think it would be good for us to describe those three things in these comments so that other programmers looking at the tests can quickly see what is being tested and how